### PR TITLE
Integration specs for views

### PR DESCRIPTION
**Pull request by Christabelle143**
# In this project i completed the following task;
#### n+1
- Make sure that the N+1 problem is solved when fetching all posts and their comments for a user. This is the corresponding wireframe view

#### Integration specs
-  Use Capybara to write integration tests for each view in your project.
  
- User index page:
  - I can see the username of all other users.
  - I can see the profile picture for each user.
  - I can see the number of posts each user has written.
  - When I click on a user, I am redirected to that user's show page.
  
- user show page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see the user's bio.
  - I can see the user's first 3 posts.
  - I can see a button that lets me view all of a user's posts.
  - When I click a user's post, it redirects me to that post's show page.
  - When I click to see all posts, it redirects me to the user's post's index page.

- User post index page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see a post's title.
  - I can see some of the post's body.
  - I can see the first comments on a post.
  - I can see how many comments a post has.
  - I can see how many likes a post has.
  - I can see a section for pagination if there are more posts than fit on the view.
  - When I click on a post, it redirects me to that post's show page.

- Post show page:
  - I can see the post's title.
  - I can see who wrote the post.
  - I can see how many comments it has.
  - I can see how many likes it has.
  - I can see the post body.
  - I can see the username of each commentor.
  - I can see the comment each commentor left.

### CODE REVIEW

File: spec/views/users/index.html.erb_spec.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    @photo = 'https://emojipedia-us.s3.dualstack.us-west-1.amazonaws.com/thumbs/160/microsoft/58/ballot-box-with-x_2612.png'
+
+    @user1 = User.create!(name: 'John Doe', bio: 'This is John Doe bio', photo: @photo, postsCounter: 16)
+    @user2 = User.create!(
+      name: 'Michal Smith', bio: 'This is Michal Smith bio', photo: @photo, postsCounter: 3
+    )
+    visit users_path
+  end
+
```
Comment (tufoinnkuo10) [inline spec/views/users/index.html.erb_spec.rb:14]: - [ ] Great job. please kindly add Tests that asserts you can see the username of all other users.

File: spec/views/users/show.html.erb_spec.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  it 'Should see the user\'s profile picture' do
+    image_src = page.find('img')['src']
+    expect(image_src).to eql @photo
+  end
+
+  it 'Should see the user\'s username' do
+    expect(page).to have_content 'Michal Smith'
+  end
+
```
Comment (tufoinnkuo10) [inline spec/views/users/show.html.erb_spec.rb:25]: - [ ] nice work. please in the show page, kindly add Tests that asserts you can see the number of posts the user has written.

File: spec/views/users/show.html.erb_spec.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  it 'Should see the user\'s first 3 posts.' do
+    expect(page).to have_content 'lorem_one'
+    expect(page).to have_content 'lorem_two'
+    expect(page).to have_content 'lorem_one'
+  end
+
+  it 'I can see a button that lets me view all of a user\'s posts.' do
+    expect(page).to have_content 'See all posts'
+  end
+
```
Comment (tufoinnkuo10) [inline spec/views/users/show.html.erb_spec.rb:39]: - [ ] please also add Tests that redirects to the post's show page when you click on a user's post

File: spec/views/posts/index.html.erb_spec.rb [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    it 'some post\'s title.' do
+      expect(page).to have_content 'Education'
+      expect(page).to have_content 'Sport'
+    end
+
+    it 'some of the post\'s body.' do
+      expect(page).to have_content 'This is first Educat post!'
+      expect(page).to have_content 'This is second Sport post!'
+    end
+
```
Comment (tufoinnkuo10) [inline spec/views/posts/index.html.erb_spec.rb:63]: - [ ] nice work. please kindly add Tests that asserts you can see the first comments on a post

File: spec/views/posts/index.html.erb_spec.rb [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+      expect(page).to have_content('Comments: 2')
+    end
+
+    it 'how many likes a post has' do
+      expect(page).to have_content('Likes: 2')
+    end
+
+    it 'a section for pagination if there are more posts than fit on the view.' do
+      expect(page).to have_content('Add Post')
+    end
```
Comment (tufoinnkuo10) [inline spec/views/posts/index.html.erb_spec.rb:74]: - [ ] kindly add Tests that redirects to the post's show page when you click on a post.

File: spec/views/posts/show.html.erb_spec.rb [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+      expect(page).to have_content 'Likes: 1'
+    end
+
+    it 'the post body.' do
+      expect(page).to have_content 'This is second Sport post!'
+    end
+
+    it 'the username of each commentor.' do
+      expect(page).to have_content 'John Doe'
+    end
```
Comment (tufoinnkuo10) [inline spec/views/posts/show.html.erb_spec.rb:63]: - [ ] please kindly add Tests that asserts you can see the comment each commentor left.

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
Hi @Christabelle143 

There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight
✅ You used the correct Github flow
✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally

Required Changes ♻️
Check the comments under the review

tufoin

Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: spec/views/users/show.html.erb_spec.rb (hunk 3 above)
Comment (Christabelle143) [reply spec/views/users/show.html.erb_spec.rb:39]: Thanks for the review @tufoinnkuo10 
This test has already been inmplemented

**Review by Munsa1 — APPROVED** [review]
Hi @Christabelle143, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**